src/passes/transform/clock_gate.cpp

```diff
@@ -0,0 +1,123 @@
+#include "coreir/passes/transform/clock_gate.h"
+
+std::string CoreIR::Passes::ClockGate::ID = "clock_gate";
+
+namespace {
+    struct CEInfo {
+        bool can_replace = false;
+        CoreIR::Instance* reg = nullptr;
+        CoreIR::Instance* mux = nullptr;
+        int mux_in_port = 0;
+        CEInfo() : can_replace(false) {}
+        CEInfo(CoreIR::Instance* reg, CoreIR::Instance* mux, bool mux_in_port) :
+            can_replace(true), reg(reg), mux(mux), mux_in_port(mux_in_port) {}
+    };
+
+    //Checks if inst of generated module and generated module has the following name
+    bool instance_of(CoreIR::Instance* inst, std::string ns, std::string name) {
```

Comment:
these look generic, can we pull them out?

Reply:
Some of them are definitely generic. I want to make a separate PR for a bunch of these helper APIs